doc/programming_guide/event_loop.rst

```diff
@@ -5,34 +5,154 @@ Introduction
-    last ``on_update`` event.
+^^^^^^^^^^^^^^^^^^^^
+is provided to update state which needs to happen at a roughly regular interval.
+The update event is not strictly paired to the draw event, but they share the same
+thread. This can cause a bottle-neck if one is significantly slower than the other.
+The event also provides a ``delta_time`` argument which is the time elapsed since the
+last ``on_update`` event. You can change the rate at which ``on_update`` is called with
+the ``update_rate`` argument when initialising your :py:class:`arcade.Window`.
 
 :py:func:`on_fixed_update`
-    is provided to update state which must happen with an exactly regular interval.
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+is provided to update state which must happen with an exactly regular interval.
+Because Arcade can't ensure the event is actually fired regularly it stores how
+much time has passed since the last update, and once enough time has passed it
+releases an ``on_fixed_update`` call. The fixed update always provides the same
+``delta_time`` argument. You can change the rate at which ``on__fixed_update`` is 
+called with the ``fixed_rate`` argument when initialising your :py:class:`arcade.Window`.
 
-**TODO**: add note about camera state resetting once that's in
+Time
+----
+While pyglet does provide a clock for scheduling events it is closely tied
+to the window's own events. For simple time keeping arcade provides global
+clock objects. Both clocks can be imported from ``arcade.clock`` as 
+``GLOBAL_CLOCK`` and ``GLOBAL_FIXED_CLOCK``
 
-All three methods are exposed to be overridden in :py:class:`arcade.Window`
-and :py:class:`arcade.View`. You may also register your own handlers
-to these events using :py:func:`arcade.Window.push_handlers`, but this is
-not recommended for beginners. 
+:py:class:`arcade.Clock`
+^^^^^^^^^^^^^^^^^^^^^^^^
+The base arcade clock tracks the elapsed time in seconds, the total number
+of clock ticks, and the amount of time that elapsed since the last tick.
+The currently active window automatically ticks the ``GLOBAL_CLOCK`` every ``on_update``.
+This means there is no reason to manually tick it. If you need more
+clocks, possibly ticking at a different rate, an :py:class:`arcade.Clock`
+can be created on the fly.
 
-Time
-====
\ No newline at end of file
+:py:class:`arcade.FixedClock`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The fixed clock tracks the same values as the normal clock, but has two special features.
+Firstly it inforces that the ``delta_time`` passed into its ``tick`` method is always the same.
+This is because advanced physics engines require consistent time. Secondly the fixed clock
+requires a sibling regular clock. It uses this clock to track how offset from the true time it is.
+Like the regular clock you may make a new :py:class:`arcade.FixedClock` at any time,
+but ensure they have a sibling.
+
+Up Coming
+^^^^^^^^^
+In future versions of arcade there will be a :py:class:`arcade.SubClock`. The sub clock will
```

Comment:
Not related to the docs here, is a second class required or would it also be possible to combine clocks, so any clock can have sub clocks as child? Just a thought, I did not look into the code.

Reply:
You're right this should really be one class It was named differently for clarity, but I will update the docs to better reflect this.